src/bot/commands/lastfm/fm.ts

```diff
@@ -249,32 +289,100 @@ export class LastFm extends Command {
                             files: [await image.getBufferAsync(Jimp.MIME_PNG)]
                         });
                     });
+                }  else {
+                    input.channel.send('Please connect your lastfm account using `lastfm set <username>`');
                 }
                 break;
             case 'artistchart':
+                if (user && user != '') {
+                    let currentTime = (Math.floor(new Date().getTime()/1000.0));
```

Comment:
It looks like you could move at least everything from line 298 to line 320 to a separate method (maybe even more that I'm missing), then recycle it in other cases where the same code is used, such as for "case: 'albumchart'". This would make the process of revising this code easier if it ever needs to be changed, since you would only need to change it in one location.